packages/client/hmi-client/src/services/model-templating.ts

```diff
@@ -20,15 +21,18 @@ export const modelTemplateOptions = [
 ].map((modelTemplate: any) => {
 	// TODO: Add templateCard attribute to Model later
 	modelTemplate.metadata.templateCard = {
-		id: -1,
+		id: modelTemplate.header.name,
 		name: modelTemplate.header.name,
 		x: 0,
 		y: 0
-	};
+	} as ModelTemplateCard;
 	return modelTemplate;
 });
 
-export function emptyModelTemplateEditor(name = 'name', description = '') {
+export function initializeModelTemplateEditor(model: Model | null = null) {
+	const name = model?.header?.name ?? 'name';
+	const description = model?.header?.description ?? 'description';
+
```

Comment:
Calling something an `Editor` suggest that it performs some actions that you can invoke, this is just a data structure. For example we didn't call "Workflow" a "WorkflowEditor" because we choose to let the WorkflowService perform the actions rather than the Workflow object itself.

Are you planning to add functionalities directly into this object?

Reply:
Not planning to add functionality I was just having trouble naming it. I changed it to `ModelTemplates` now